src/qtism/runtime/tests/RouteItem.php

```diff
@@ -426,4 +450,69 @@ public function getTimeLimits($excludeItem = false): RouteTimeLimitsCollection
 
         return $timeLimits;
     }
+
+    public function getEffectiveBranchRules(): BranchRuleCollection
+    {
+        if ($this->getBranchRules()->count() > 0) {
+            return $this->getBranchRules();
+        }
+
+        $sectionBranchRules = $this->getEffectiveSectionBranchRules();
+
+        if ($sectionBranchRules === null || $sectionBranchRules->count() > 0) {
+            return $sectionBranchRules ?? new BranchRuleCollection();
+        }
+
+        $testPart = $this->getTestPart();
+        $currentItemSections = $this->getAssessmentSections()->getArrayCopy();
+
+        if (!$testPart->isLastSection($currentItemSections[0])) {
+            return new BranchRuleCollection();
+        }
+
+        return $testPart->getBranchRules();
+    }
+
+    /**
+     * Selects branching rules from the section/subsection.
+     * Branching rules will be selected only if the item or subsection is the last one in the parent section.
+     *
+     * @return BranchRuleCollection|null Returns the branching rules for the last section or null if the
+     *                                   element/subsection is not the last.
+     */
+    private function getEffectiveSectionBranchRules(): ?BranchRuleCollection
+    {
+        /** @var AssessmentSection[] $sections */
+        $sections = $this->getAssessmentSections()->getArrayCopy();
+
+        // Remove the current section from the section list, as this section contains a list of items, not sections.
+        $currentSection = array_pop($sections);
+
+        if ($currentSection === null || !$currentSection->isLastSectionPart($this->getAssessmentItemRef())) {
+            return null;
```

Comment:
Maybe, can we return a collection here to avoid comparison with `null` and count above?

```suggestion
            return new BranchRuleCollection();
```

Reply:
Done: https://github.com/oat-sa/qti-sdk/compare/12bc4edc855ef96737f9c4c0f1bb68d298c85e9d...045e56cea97d32d51d3cf8bca41936c4bfbee66c